scripts/Verify-GoModules.ps1

```diff
@@ -0,0 +1,63 @@
+function newTemporaryDirectory {
+    $parent = [System.IO.Path]::GetTempPath()
+    [string] $name = [System.Guid]::NewGuid()
+    New-Item -ItemType Directory -Path (Join-Path $parent $name)
+}
+
+function updateVendor {
+    param (
+        [string]$path
+    )
+    $currentPath = (Get-Location).Path
+    Set-Location $path
+    go mod vendor
+    go mod tidy 
+    Set-Location $currentPath
+}
+
+function matchingHashes {
+    param (
+        [string]$rootPath,
+        [string]$tempPath
+    )
+    $rootHashes = Get-ChildItem -Recurse -Path $rootPath | foreach  {Get-FileHash -Path $_.FullName -Algorithm SHA256}
+    if (-not $?) {
+        return $false
+    }
+    $tempHashes = Get-ChildItem -Recurse -Path $tempPath | foreach  {Get-FileHash -Path $_.FullName -Algorithm SHA256}
+    if (-not $?) {
+        return $false
+    }
+    $diff = Compare-Object -ReferenceObject $rootHashes.Hash -DifferenceObject $tempHashes.Hash
+    if ($diff.Count -eq 0) {
+        return $true
+    }
+    return $false
+}
+
+$rootPath = $args[0]
+$subdir = $args[1]
+
+$tempDir = newTemporaryDirectory
+Copy-Item -Path $rootPath/* -Destination $tempDir -Recurse
+if (-not $?) {
+    Remove-Item $tempDir -Recurse
+    exit 1
+}
+
+updateVendor $tempDir/$subdir
+if (-not $?) {
+    Remove-Item $tempDir -Recurse
+    exit 1
+}
+
+$hashesMatch = matchingHashes $rootPath/$subdir/vendor $tempDir/$subdir/vendor
+if (-not $?) {
+    Remove-Item $tempDir -Recurse
+    exit 1
+}
+
+Remove-Item $tempDir -Recurse
+if ($hashesMatch -ne $true) {
+    exit 1
+}
```

Comment:
It'd be nice to output what files were the offenders/didn't match from the pr vs. doing it here. Or a prompt to display what to run to potentially remedy the problem

Reply:
The issue is that this script needs to be run with admin privileges, which makes grabbing the output from this script a little troublesome. I could add an error message in the CI job that mentions to run `go mod vendor; go mod tidy` on failure. Does that work?

Reply:
Sure, as long as the error message is tailored for go mod vendor in the root vs /test (e.g. if the test modules validation fails the error msg should mention to vendor+tidy in test).

Reply:
Updated. PTAL.